search/cloud-function/README.md

```diff
@@ -0,0 +1,124 @@
+
+If you want even more control see
+[Vertex AI APIs for building search and RAG experiences](https://cloud.google.com/generative-ai-app-builder/docs/builder-apis),
+but first explore the out-of-the-box offering because it's easy to setup.
+
+This example is for the out-of-the-box Vertex AI Search supporting many
+configurations and data types.
+
+## Pre-requisites
+
+Before you can use these functions to query Vertex AI Search, you need to create
+and populate a search "data store"; read through instructions in
+[get started with generic search](https://cloud.google.com/generative-ai-app-builder/docs/try-enterprise-search).
+These functions could easily be adapted to other types of Vertex AI Search like
+[generic recommendations](https://cloud.google.com/generative-ai-app-builder/docs/try-generic-recommendations),
+[media search](https://cloud.google.com/generative-ai-app-builder/docs/try-media-search),
+[media recommendations](https://cloud.google.com/generative-ai-app-builder/docs/try-media-recommendations),
+[healthcare search](https://cloud.google.com/generative-ai-app-builder/docs/create-data-store-hc),
+or even
+[retail product discovery](https://cloud.google.com/solutions/retail-product-discovery#documentation).
+
+You'll need to collect the following details from your search app data store:
+
+```python
+PROJECT_ID = "YOUR_PROJECT_ID"  # alphanumeric
+LOCATION = "global"  # or an alternate location
+DATA_STORE_ID = "YOUR_DATA_STORE_ID"  # not the app id, alphanumeric
+```
+
+Additionally you'll need to keep track of some of the choices you make when you
+configure Vertex AI Search.
+
+### Type of data source
+
+- UNSTRUCTURED
+- STRUCTURED
+- WEBSITE
+- BLENDED
+
+```python
+ENGINE_DATA_TYPE = UNSTRUCTURED
+```
+
+### Type of chunks to return
+
+- DOCUMENT_WITH_SNIPPETS
+- DOCUMENT_WITH_EXTRACTIVE_SEGMENTS
+- CHUNK
+- NONE
+
+```python
+ENGINE_CHUNK_TYPE = DOCUMENT_WITH_EXTRACTIVE_SEGMENTS
+```
+
+### Type of summarization
+
+- NONE results only
+- VERTEX_AI_SEARCH LLM add on provided by
+[Vertex AI Search](https://cloud.google.com/generative-ai-app-builder/docs/enterprise-search-introduction)
+<!-- NOT ready yet
+- GENERATE_GROUNDED_ANSWERS use the
+  [Generate grounded answers with RAG](https://cloud.google.com/generative-ai-app-builder/docs/grounded-gen)
+  provided by
+  [Vertex AI Search Builder APIs](https://cloud.google.com/generative-ai-app-builder/docs/builder-apis)
+- GEMINI use one of the Gemini models to generate an answer from the results -->
+
+```python
+SUMMARY_TYPE = VERTEX_AI_SEARCH
+```
+
+## Architecture
+
+1. Vertex AI Search is an API hosted on Google Cloud
+2. You will call that API via a Google Cloud Function, which exposes its own API
+3. Your users will the Google Cloud Function API, via your custom app or UI
+
+```mermaid
+flowchart LR
+    A[fa:fa-search Vertex Search AI] --> B(Google Cloud Function)
+    B --> C[My App Server]
+    C -->|One| D[fa:fa-laptop web]
+    C -->|Two| E[fa:fa-mobile mobile]
+```
```

Comment:
This is so neat! I've never seen this text based flowchart used in GitHub markdown.